### ISSUE TITLE
impl Render for borrows ? 

### ISSUE THREAD

**Issue by RCasatta** [lambda-fairy/maud#266]
I struggled a bit understanding why this code doesn't work

```
#[test]
fn yet_another_test() {
    let mut vec = vec![];
    let a = html! { div { "foo" } };
    vec.push(a);
    let b = html! { div { "bar" } };
    vec.push(b);
    dbg!(&vec);

    let f = html! { @for el in &vec { (el) } };
    assert_eq!("<div>foo</div><div>bar</div>", f.into_string());
}
```

before realizing I had to Dereference el -> `*el`

Maybe it's possible to impl Render for borrowed values ?

**Comment by lambda-fairy** [comment]
That's a good point.

I think it would be possible to add an impl for `&PreEscaped<T>`.

But I don't think we can generalize beyond that. An impl for all `&T` will conflict with the existing [blanket impl for `T: Display`](https://github.com/lambda-fairy/maud/blob/8990049101246aec0cf3c784182ca819c2c54ebc/maud/src/lib.rs#L72), because `Display` is implemented for all `&T` already.

I think the ideal solution is to remove the blanket impl for `T: Display`. This is what markup.rs and sailfish do. But we'll need to check how much code that breaks.

**Comment by RCasatta** [comment]
> I think it would be possible to add an impl for &PreEscaped<T>.

this would be nice since it will cover most common case I think

**Comment by lambda-fairy** [comment]
This is fixed by #320
